task/10-katas-1-tasks.js

```diff
@@ -16,9 +16,90 @@
+            let prevDirection = sides[(sides.indexOf(firstChar)+3)%4];
+            let nextDirection = sides[(sides.indexOf(firstChar)+1)%4];
+            return new EndBuilder().build(firstChar+'b', 'b')
+            .concat([new CompasPoint(firstChar+nextDirection+firstChar, 
+                                     sides.indexOf(firstChar)*90+22.5*this.countSign(firstChar, nextDirection))])
+            .concat([new CompasPoint(firstChar+prevDirection+firstChar, 
+                                     sides.indexOf(firstChar)*90+22.5*this.countSign(firstChar, prevDirection))])
+        }
+        
+        countSign(firstChar, curChar){
+            return Math.sign((sides.indexOf(curChar)+sides.indexOf(firstChar))%4-(sides.indexOf(curChar)+sides.indexOf(firstChar)+2)%4);
+        }
+        
+    }
+
+    class EndBuilder{
+        build(begin, endChar){
+            let startCharIndex = sides.indexOf(begin.charAt(0))
+            let prevDirection = sides[(sides.indexOf(endChar)+3)%4];
+            let nextDirection = sides[(sides.indexOf(endChar)+1)%4];
+            let prevDirectionForb = sides[(startCharIndex+3)%4];
+            let nextDirectionForb = sides[(startCharIndex+1)%4];
+            return begin.charAt(0) == begin.charAt(1) ? [new CompasPoint(begin+nextDirection, 
+                                                                         startCharIndex*90 + 22.5*this.countSign(begin.charAt(0), nextDirection)), 
+                                                        new CompasPoint( begin+prevDirection,
+                                                                         startCharIndex*90 + 22.5*this.countSign(begin.charAt(0), prevDirection))]:
+                    endChar == 'b' ? [new CompasPoint( begin+nextDirectionForb, 
+                                                       startCharIndex*90 + 11.25*this.countSign(begin.charAt(0), nextDirectionForb)), 
+                                    new CompasPoint( begin+prevDirectionForb,
+                                                       startCharIndex*90 + 11.25*this.countSign(begin.charAt(0), prevDirectionForb))]:
+                                    
+                                    [new CompasPoint( begin, 90*startCharIndex+45*this.countSign(begin.charAt(0), begin.charAt(1))),
+                                    new CompasPoint(begin+'b'+begin.charAt(0),
+                                                    90*startCharIndex+45*this.countSign(begin.charAt(0), begin.charAt(1)) 
+                                                        + 11.25*this.countHardSign(begin.charAt(0), begin.charAt(1), begin.charAt(0))), 
+                                    new CompasPoint(begin+'b'+begin.charAt(1),
+                                                    90*startCharIndex+45*this.countSign(begin.charAt(0), begin.charAt(1)) 
+                                                        + 11.25*this.countHardSign(begin.charAt(0), begin.charAt(1), begin.charAt(1)))]
```

Comment:
Please, avoid this code style, if-else is more readable than multiple conditional operators ? :